source/dub/commandline.d

```diff
@@ -848,12 +849,55 @@ class BuildCommand : GenerateCommand {
 		args.getopt("f|force", &m_force, [
 			"Forces a recompilation even if the target is up to date"
 		]);
+		args.getopt("y|yes", &m_yes, [
+			`Automatic yes to prompts. Assume "yes" as answer to all prompts and run non-interactively.`
+		]);
 		super.prepare(args);
 		m_generator = "build";
 	}
 
 	override int execute(Dub dub, string[] free_args, string[] app_args)
 	{
+		// single package files don't need to be downloaded, they are on the disk.
+		if (free_args.length >= 1 && !m_single) {
+			auto package_name = free_args[0];
+			auto pack = dub.packageManager.getFirstPackage(package_name);
+
+			if (!pack) {
```

Comment:
Switch the condition and do an early exit, if possible.

Reply:
That duplicates `return super.execute(dub, free_args, app_args);`, but no strong feelings either way. I changed it to the early exit.